osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/domain/valueobjects/DaylightSavingTimeTransition.java

```diff
@@ -207,16 +207,28 @@ public boolean isValid(final String transition) {
                 } catch (final NumberFormatException nfe) {
                     return false;
                 }
-                if (m < 1 || m > 12 || w < 1 || w > 5 || d < 0 || d > 6) {
-                    return false;
-                }
+                final boolean validRange = this.checkRange(m, w, d);
                 if (timeSeparatorPos == -1) {
                     return true;
                 }
-                return transition.length() > timeSeparatorPos
+                return validRange && transition.length() > timeSeparatorPos
                         && this.isValidTime(transition.substring(timeSeparatorPos + 1));
             }
 
+            private boolean checkRange(final int m, final int w, final int d) {
+                if (m < 1 || m > 12) {
+                    return false;
+                }
+                if (w < 1 || w > 5) {
+                    return false;
+                }
+                if (d < 0 || d > 6) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1126)